yb-voyager/cmd/importDataFileTaskImporter_test.go

```diff
@@ -278,3 +286,393 @@ func TestTaskImportErrorsOutWithAbortErrorPolicy(t *testing.T) {
+		}
+	}
+}
+
+func TestImportBatchErrorDetection_YugabyteDB(t *testing.T) {
+	ctx := context.Background()
+
+	// Setup YB container
+	ybContainer := testcontainers.NewTestContainer("yugabytedb", nil)
+	err := ybContainer.Start(ctx)
+	require.NoError(t, err, "Failed to start YugabyteDB container")
+	defer ybContainer.Terminate(ctx)
+
+	// Create target database instance
+	targetDB := createTargetYugabyteDB(t, ybContainer)
+
+	// Create Voyager schema and metadata tables
+	err = targetDB.CreateVoyagerSchema()
+	require.NoError(t, err, "Failed to create Voyager schema")
+
+	// Create test table
+	createTestTableWithConstraints(t, ybContainer)
+
+	// Run test cases
+	testCases := createTestCases()
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			testErrorDetection(t, targetDB, tc)
+		})
+	}
+}
+
+func TestImportBatchErrorDetection_PostgreSQL(t *testing.T) {
+	ctx := context.Background()
+
+	// Setup PG container
+	pgContainer := testcontainers.NewTestContainer("postgresql", nil)
+	err := pgContainer.Start(ctx)
+	require.NoError(t, err, "Failed to start PostgreSQL container")
+	defer pgContainer.Terminate(ctx)
+
+	// Create target database instance
+	targetDB := createTargetPostgreSQL(t, pgContainer)
+
+	// Create Voyager schema and metadata tables
+	err = targetDB.CreateVoyagerSchema()
+	require.NoError(t, err, "Failed to create Voyager schema")
+
+	// Create test table
+	createTestTableWithConstraints(t, pgContainer)
+
+	// Run test cases
+	testCases := createTestCases()
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			testErrorDetectionPG(t, targetDB, tc)
+		})
+	}
+}
+
+func createTargetPostgreSQL(t *testing.T, container testcontainers.TestContainer) *tgtdb.TargetPostgreSQL {
+	// Get container connection details
+	host, port, err := container.GetHostPort()
+	require.NoError(t, err, "Failed to get container host/port")
+
+	config := container.GetConfig()
+
+	// Create TargetConf
+	tconf := &tgtdb.TargetConf{
+		Host:         host,
+		Port:         port,
+		User:         config.User,
+		Password:     config.Password,
+		DBName:       config.DBName,
+		Schema:       "public", // Set default schema
+		TargetDBType: tgtdb.POSTGRESQL,
+	}
+
+	// Create and initialize target database
+	targetDB := tgtdb.NewTargetDB(tconf).(*tgtdb.TargetPostgreSQL)
+	err = targetDB.Init()
+	require.NoError(t, err, "Failed to initialize target database")
+
+	// Initialize connection pool
+	err = targetDB.InitConnPool()
+	require.NoError(t, err, "Failed to initialize connection pool")
+
+	return targetDB
+}
+
+func testErrorDetectionPG(t *testing.T, targetDB *tgtdb.TargetPostgreSQL, tc testCase) {
+	// Create batch with erroneous data
+	tableName := sqlname.NameTuple{CurrentName: sqlname.NewObjectName(tgtdb.POSTGRESQL, "test_schema", "test_schema", "error_test_table")}
+	batch := createBatchFromData(t, tc.dataGenerator(), tableName)
+
+	// Create import batch args
+	args := &tgtdb.ImportBatchArgs{
```

Comment:
ideally, your `createBatchFromData` should return the batch args as well, because it knows what is being created (the columns, the format, the delimiter, etc)